src/shared/calendarspec.c

```diff
@@ -888,11 +892,14 @@ int calendar_spec_from_string(const char *p, CalendarSpec **spec) {
         assert(p);
         assert(spec);
 
-        c = new0(CalendarSpec, 1);
+        c = new(CalendarSpec, 1);
         if (!c)
                 return -ENOMEM;
-        c->dst = -1;
-        c->timezone = NULL;
+
+        *c = (CalendarSpec) {
+                .dst = -1,
+                .timezone = NULL,
```

Comment:
actually, this last line can be dropped, after all during structured initialization pointers initialize to NULL anyway..

Reply:
ouch.